docs/dev/guidelines/client-design.rst

```diff
@@ -0,0 +1,241 @@
+
+            // My Special Component
+            $my-special-component-my-box-background-color: $gray-600;
+
+        .. IMPORTANT::
+            The two options below are meant as a fallback for special cases.
+            Please justify the use of these options in your PR description.
+
+        If you need a **slightly modified preset color for some reason**, use `SCSS functions <https://sass-lang.com/documentation/modules/color>`_ to derive it.
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: darken($success, 20);
+
+        **If all of this doesn't work, define your own colors.**
+
+        .. code-block:: scss
+
+            // My Special Component
+            $my-special-component-my-box-background-color: #123456;
+
+1. Theme-specific global styles
+===============================
+
+It might happen that you need to modify a global style rule in one of the themes, for example if you're using a external library which styles need to be overridden.
+
+| Each theme has a dedicated file custom global styles can be added to: ``theme-dark.scss`` and ``theme-default.scss``.
+| For styles that should be applied in both themes, use ``global.scss``.
+
+2. Theme Service
+================
+
+There will be occasions where you need to know in your components which theme is currently applied.
+The ``ThemeService`` will provide this information where needed.
+For example, you could add a reactive flag to your component that indicates whether or not the current theme is dark:
+
+.. code-block:: ts
+
+    @Component({
+        selector: 'jhi-my-component',
+        templateUrl: './my-component.component.html',
+        styleUrls: ['my-component.component.scss'],
+    })
+    export class MyComponent implements OnInit, OnDestroy {
+        isDark = false;
+        themeSubscription: Subscription;
+
+        ngOnInit() {
+            this.themeSubscription = this.themeService.getCurrentThemeObservable().subscribe((theme) => {
+                this.isDark = theme === Theme.DARK;
+            });
+        }
+
+        ngOnDestroy() {
+           this.themeSubscription.unsubscribe();
+        }
+    }
+
+| From there, you can do whatever you need to do to change the behavior of your component based on the theme.
+| Alternatively, you can execute any actions directly in the ``subscribe()`` block.
+  **The service will fire an event containing the current theme immediately as soon as you subscribe**, and one more
+  event for each theme change from then on util you unsubscribe.
+| You can get the current theme using ``themeService.getCurrentTheme()`` at any time.
+
+
+Additionally, it's possible to change the theme programmatically. **However, this should be rare**: Usually, the user decides which theme
+they want to use by using the theme switching component in the navbar. Any use of this must therefore be justified and
+survive a review by @bassner ;)
```

Comment:
Should we really write in the documentation that you personally will review that? I'd expect you to leave the development-team after some time, maybe we should just write "survive a detailed review"?

Reply:
xD yeah maybe I just pasted my thoughts there lol